### PR TITLE
log15 migration: migrate log15 usage in frontend/backend (+friends) to sourcegraph/log

### PR DESCRIPTION
**Pull request by burmudar**
This started out with aiming to just migrate the log15 usage in frontend/backend to use sourcegraph/log and then the log propagation spread out to touch a log more

## Test plan
Uni tests

### CODE REVIEW

File: cmd/frontend/backend/repos.go [hunk 1]
```diff
@@ -222,7 +224,7 @@ func (s *repos) GetInventory(ctx context.Context, repo *types.Repo, commitID api
 	ctx, cancel := context.WithTimeout(ctx, 3*time.Minute)
 	defer cancel()
 
-	invCtx, err := InventoryContext(repo.Name, s.db, commitID, forceEnhancedLanguageDetection)
+	invCtx, err := InventoryContext(s.logger.Scoped("InventoryContext", ""), repo.Name, s.db, commitID, forceEnhancedLanguageDetection)
```
Comment (sashaostrikov) [inline cmd/frontend/backend/repos.go:227]: nit: we can add a description here (instead of `""`) which can be similar to `InventoryContext` function doc

Comment (sashaostrikov) [reply cmd/frontend/backend/repos.go:227]: same thing can apply to other logger instantiations in this PR

**Review by sashaostrikov — APPROVED** [review]
LGTM!

File: cmd/frontend/backend/repos.go (hunk 1 above)
Comment (burmudar) [reply cmd/frontend/backend/repos.go:227]: Will do!

**Review by unknwon — APPROVED** [review]

File: cmd/frontend/graphqlbackend/search_results_stats_languages.go [hunk 2]
```diff
@@ -29,7 +29,8 @@ func (srs *searchResultsStats) Languages(ctx context.Context) ([]*languageStatis
 		return nil, err
 	}
 
-	langs, err := searchResultsStatsLanguages(ctx, srs.logger, srs.sr.db, matches)
+	logger := srs.logger.Scoped("languages", "provide stats on langauge from the search results")
```
Comment (burmudar) [inline cmd/frontend/graphqlbackend/search_results_stats_languages.go:32]:
```suggestion
	logger := srs.logger.Scoped("languages", "provide stats on langauges from the search results")
```

File: cmd/frontend/graphqlbackend/set_external_service_repos.go [hunk 3]
```diff
@@ -82,7 +82,7 @@ func (r *schemaResolver) SetExternalServiceRepos(ctx context.Context, args struc
 		return nil, err
 	}
 
-	if err = backend.SyncExternalService(ctx, es, 5*time.Second, r.repoupdaterClient); err != nil {
+	if err = backend.SyncExternalService(ctx, r.logger.Scoped("SyncExternalService", ""), es, 5*time.Second, r.repoupdaterClient); err != nil {
```
Comment (burmudar) [inline cmd/frontend/graphqlbackend/set_external_service_repos.go:85]:
```suggestion
	if err = backend.SyncExternalService(ctx, r.logger, es, 5*time.Second, r.repoupdaterClient); err != nil {
```

File: cmd/frontend/backend/external_services.go [hunk 4]
```diff
@@ -143,7 +144,7 @@ func SyncExternalService(ctx context.Context, svc *types.ExternalService, timeou
 		// If context error is anything but a deadline exceeded error, we do not want to propagate
 		// it. But we definitely want to log the error as a warning.
 		if ctx.Err() != nil && ctx.Err() != context.DeadlineExceeded {
-			log15.Warn("SyncExternalService: context error discarded", "err", ctx.Err())
+			logger.Warn("SyncExternalService: context error discarded", log.Error(ctx.Err()))
```
Comment (bobheadxi) [inline cmd/frontend/backend/external_services.go:147]:
```suggestion
			logger.Warn("context error discarded", log.Error(ctx.Err()))
```

File: cmd/frontend/backend/repos_test.go [hunk 5]
```diff
@@ -235,8 +235,6 @@ func TestReposGetInventory(t *testing.T) {
 
 func TestMain(m *testing.M) {
 	flag.Parse()
-	if !testing.Verbose() {
-		log15.Root().SetHandler(log15.DiscardHandler())
-	}
+	logtest.Init(m)
```
Comment (bobheadxi) [inline cmd/frontend/backend/repos_test.go:238]: We should preserve the silencing behaviour here:

```suggestion
	if !testing.Verbose() {
		logtest.InitWithlevel(m, log.LevelNone)
	} else {
		logtest.Init(m)
	}
```

File: enterprise/cmd/frontend/internal/auth/gitlaboauth/config.go [hunk 6]
```diff
@@ -49,7 +50,7 @@ type Provider struct {
 	providers.Provider
 }
 
-func parseConfig(cfg conftypes.SiteConfigQuerier, db database.DB) (ps []Provider, problems conf.Problems) {
+func parseConfig(cfg conftypes.SiteConfigQuerier, logger log.Logger, db database.DB) (ps []Provider, problems conf.Problems) {
```
Comment (bobheadxi) [inline enterprise/cmd/frontend/internal/auth/gitlaboauth/config.go:53]: Logger should be the first parameter:

```suggestion
func parseConfig(logger log.Logger, cfg conftypes.SiteConfigQuerier, db database.DB) (ps []Provider, problems conf.Problems) {
```

File: enterprise/cmd/frontend/internal/auth/oauth/session.go [hunk 7]
```diff
@@ -36,13 +36,14 @@ type SessionIssuerHelper interface {
 	SessionData(token *oauth2.Token) SessionData
 }
 
-func SessionIssuer(db database.DB, s SessionIssuerHelper, sessionKey string) http.Handler {
+func SessionIssuer(logger log.Logger, db database.DB, s SessionIssuerHelper, sessionKey string) http.Handler {
+	logger = logger.Scoped("SessionIssuer.HandlerFunc", "validates a token and then sets up a session")
```
Comment (bobheadxi) [inline enterprise/cmd/frontend/internal/auth/oauth/session.go:40]: If there's the issue itself is a handler, I think just noting the scope as SessionIssuer is sufficient:

```suggestion
	logger = logger.Scoped("SessionIssuer", "validates a token and then sets up a session")
```

File: enterprise/cmd/frontend/internal/auth/gitlaboauth/config.go (hunk 6 above)
Comment (burmudar) [reply enterprise/cmd/frontend/internal/auth/gitlaboauth/config.go:53]: aaaah I was unsure about this! Thanks :D

File: enterprise/cmd/frontend/internal/auth/oauth/session.go (hunk 7 above)
Comment (burmudar) [reply enterprise/cmd/frontend/internal/auth/oauth/session.go:40]: Yeah good point. I was inbetween minds about this but ultimately agree!

**Review by bobheadxi — APPROVED** [review]
Nice! 😁

**Pull request merged**